src/main/java/com/amazon/ion/impl/bin/IonRawBinaryWriter_1_1.kt

```diff
@@ -366,25 +423,96 @@ class IonRawBinaryWriter_1_1 internal constructor(
 
     override fun writeBool(value: Boolean) = writeScalar { writeBoolValue(buffer, value) }
 
-    override fun writeInt(value: Long) = writeScalar { writeIntValue(buffer, value) }
-
-    override fun writeInt(value: BigInteger) = writeScalar { writeIntValue(buffer, value) }
-
-    override fun writeFloat(value: Float) = writeScalar { writeFloatValue(buffer, value) }
-
-    override fun writeFloat(value: Double) = writeScalar { writeFloatValue(buffer, value) }
+    override fun writeInt(value: Long) = writeTaggedOrTaglessScalar(
+        ifTagged = { writeIntValue(buffer, value) },
+        ifTagless = { primitiveType ->
+            when (primitiveType) {
+                // TODO: Do we need to check bounds?
```

Comment:
I think so... also, how do we differentiate between signed / unsigned?

Reply:
I wasn't sure if it would be a performance issue or not to check the bounds.

> how do we differentiate between signed / unsigned?

Well, there's not really a difference when you're truncating an integer to e.g. 8 bits. As long as it's in the range of acceptable values for the target size, then it will be encoded correctly. That's why we're able to have `writeFixedIntOrUInt(...)` because the only differentiator between how sign and unsigned fixed ints are written is calculating the number of bytes needed to represent the value.

Reply:
For these general-purpose int writing APIs, I feel like we have to have bounds checking for fixed-width tagless types as a safety measure. However, I could imagine also providing additional APIs specifically for writing the fixed-width tagless types, e.g. `writeTaglessInt32(value: Int)`, `writeTaglessUInt8(value: Byte)` that could sidestep the validation because the argument type's size is guaranteed to fit in the size of the requested tagless type. Since all Java integer primitives are signed, we'd have to specify as part of the API that we do something like the following for the UInts:
```
void writeTaglessUInt8(byte[] value) {
   writeIntValue(buffer, value & 0xFF);
}
```

Reply:
Note: this doesn't need to block this PR.